tools/cli/main.js

```diff
@@ -745,6 +745,35 @@ Fiber(function () {
     // It is a plain old argument!
     rawArgs.push(term);
   }
+  
+  // Check if meteor is run as root only on UNIX platforms
+  var unixPlatforms = { 
+    'darwin': true,
+    'linux': true,
+    'freebsd': true
+  };
+  var platform = process.platform;
+
+  if (_.has(unixPlatforms, platform)){
+    var isRoot = process.env.USERNAME == 'root';
+    var isRootSafe = _.has(rawOptions, '--unsafe-perm');
+
+    if(isRoot && !isRootSafe){
+      Console.error("");
+      Console.error("It seems that you are running meteor as root, if you really know what you are doing, add the unsafe-perm flag.");
+      Console.error("");
+      Console.error("meteor --unsafe-perm <command>");
+      Console.error("");
+      process.exit(1);
+    }
+    if (isRoot && isRootSafe) {
+      Console.info("");
+      Console.info("You are running meteor as root, don't forget to chown your .meteor directory back to yourself.");
+      Console.info("");
+      Console.info("chown -Rh <username> ~/.meteor");
```

Comment:
If run as root (with `sudo` or not) it would still use `$HOME/.meteor` (which in most cases would be `/root`) for `meteor-tool`-related stuff which seems like it should be fine and I don't believe is the source of the problem.

I think the issue arises with the `root`-owned files being written in `<app-dir>/.meteor/local/` which would be in the user's home directory, no?  (Though even that should be fine so long as they ALWAYS run every Meteor command with root, but they probably don't understand what they are doing if they're doing this.).